web/src/popover_menus.js

```diff
@@ -188,11 +188,54 @@ function on_show_prep(instance) {
     popovers.hide_all_except_sidebars();
 }
 
+function get_props_for_popover_centering(popover_props) {
+    return {
+        arrow: false,
+        getReferenceClientRect: () => ({
+            width: 0,
+            height: 0,
+            left: 0,
+            top: 0,
+        }),
+        popperOptions: {
+            modifiers: [
+                {
+                    name: "offset",
+                    options: {
+                        offset({popper}) {
+                            // Calculate the offset needed to place the reference in the center
+                            const x_offset_to_center = window.innerWidth / 2;
+                            const y_offset_to_center = window.innerHeight / 2 - popper.height / 2;
+
+                            return [x_offset_to_center, y_offset_to_center];
+                        },
+                    },
+                },
+            ],
+        },
+        onMount(instance) {
+            $("body").append($("<div>").attr("id", "popover-overlay"));
+            popover_props.onMount(instance);
+        },
```

Comment:
How exactly does this block work? I don't see what attaches the popover to `#popover-overlay`

Reply:
`#popover-overlay` is an empty container used for background styling. It does nothing else and is attached to the `body`.

Our popovers are also attached to the `body` element, so they are siblings with `#popover-overlay`.

Reply:
Copying and pasting the `get_props_for_popover_centering` and `show_popover_menu` in #26375 makes me think that creating `get_props_for_popover_centering` should be in a separate preparatory commit that we can merge safely.

Reply:
OK -- should this be called `popover-overlay-background` then?

Reply:
Yeah, I renamed that and also made a separate commit (f8c57ebe) to introduce `get_props_for_popover_centering`.